Docs: Explanations for missing sections (#1128)
esy/esy: docs/concepts.md

environment is reproducible and doesn't depend on the current state of package
registries (either npm or opam).

## Stores 

Esy maintains three kinds of artifacts - sources, builds and installed artfiacts - in three different locations.

### Sources

> Source entries in the esy store are source code files and the recipies that build and install them bundled together. Change any of them, and esy creates a different entry for the source of a package.

Reviewer: nit: typo
```suggestion
Source entries in the esy store are source code files and the recipes that build and install them bundled together. Change any of them, and esy creates a different entry for the source of a package.
```

If, for instance, build instructions of a package from opam are overridden (and the source files are not necessarily touched), esy places them in a different location.

> Conceptually, a source entry in the esy store is both it's source files and it's recipie.

Reviewer: nit: typo

```suggestion
Conceptually, a source entry in the esy store is both its source files and its recipe.
```

### Build store

Build store is simply location where build artifacts are stored. They can be local to a project or global (so that they can be reused).

> Build artifacts (object files, bytecode files etc) stored as separate entries dependending on the instructions that build and install them, environment variables present during its build and of course the sources (source code + recipie).

Reviewer: nit: typo
```suggestion
Build artifacts (object files, bytecode files etc) stored as separate entries dependending on the instructions that build and install them, environment variables present during its build and of course the sources (source code + recipe).
```
